classes/emailTemplate/DAO.php

```diff
@@ -235,13 +235,17 @@ public function getMainEmailTemplatesFilename()
      * skipping others
      * @param bool $skipExisting If true, do not install email templates
      * that already exist in the database
+     * @param bool $recordTemplateGroupAccess - If true, records the templates as unrestricted.
+     * By default, it is set to false to ensure compatibility with older processes (e.g., migrations)
+     * where the `email_template_user_group_access` table may not exist at the time of execution.
      *
      */
     public function installEmailTemplates(
         string $templatesFile,
         array $locales = [],
         ?string $emailKey = null,
-        bool $skipExisting = false
+        bool $skipExisting = false,
+        $recordTemplateGroupAccess = false
```

Comment:
I mention this directly below, but I don't think this optional parameter will be necessary. I think the assumption of "unrestricted if no additional data provided" below is enough.

Reply:
For posterity, we talked about it and decided to keep thing `$recordTemplateGroupAccess` parameter as described here to avoid any issues as part of the upgrade process.